dowhy/causal_estimators/propensity_score_estimator.py

```diff
@@ -68,6 +69,24 @@ def __init__(self, *args, propensity_score_model=None,
             self.logger.error(error_msg)
             raise Exception(error_msg)
 
+    def _refresh_propensity_score(self):
+        if self.recalculate_propensity_score is True:
+            if self.propensity_score_model is None:
+                self.propensity_score_model = linear_model.LogisticRegression()
+            self.propensity_score_model.fit(self._observed_common_causes, self._treatment)
+            self._data[self.propensity_score_column] = self.propensity_score_model.predict_proba(self._observed_common_causes)[:, 1]
+        else:
+            # check if user provides the propensity score column
+            if self.propensity_score_column not in self._data.columns:
+                if self.propensity_score_model is None:
+                    raise ValueError(f"""Propensity score column {self.propensity_score_column} does not exist, nor does a propensity_model. 
+                    Please specify the column name that has your pre-computed propensity score, or a model to compute it.""")
+                else:
```

Comment:
Even if the `propensity_score_model` is not None, it is possible that it is not a fitted model. For example, a user may provide `linear_model.LogisticRegression()` as the model. 
So, we will also need to check whether the `propensity_score_model` is a fitted object. 
But the expectation is that the user only provides the class of the propensity model, not the fitted object. 

I guess the benefit is a consistent interface. Then we will need to update the docs for the `propensity_score_model` parameter (it can be a fitted model if recalculate flag is False).

Reply:
Main motivation for this one is as a companion to https://github.com/py-why/dowhy/pull/485, to allow to evaluate specifically this estimator family on new datasets without refitting the `propensity_score_model`. The behavior is exactly as it was before, except in the case when the recalculate flag is `False` and there is a  non-None `propensity_score_model`. 

Or do you think that if the recalculate flag is `False`, and the `propensity_score_model` is not `None` but is not fitted, we should try to fit it first?

Reply:
> Or do you think that if the recalculate flag is False, and the propensity_score_model is not None but is not fitted, we should try to fit it first

Yes, but I suggest that we do not fit it, simply raise an error. The current API expects a user to pass an unfitted model, e.g., `LinearRegression()` and then dowhy fits the model internally. 

For estimating effect on new data, we can use the following logic: if `recalculate_propensity_score=False`, then the user should either provide a fitted model or they should provide the pre-filled propensity column. If they do not provide any of these two, then we raise an error. 

So maybe the only change needed in the PR is to add a check on whether the model is fitted in line 81: model is none or unfitted, both should raise an error.

Reply:
Sounds good. The error will be raised anyway if we try to `predict()` with an unfitted model (in fact that's the only way I can think of to check whether the model is fitted, after all we can't guarantee the model will support `sklearn.utils.validation.check_is_fitted`).
So the only thing to do is to catch exceptions when calling `predict()` and re-raise them with a more helpful message?

Reply:
That sounds good! Happy to merge once you make these changes.

Reply:
Done :)